src/ServerSide/InternalTypes.elm

```diff
@@ -105,6 +108,15 @@ decodeTagger =
         ]
 
 
+decodeKeyedNode : Json.Decode.Decoder NodeRecord
+decodeKeyedNode =
+    Json.Decode.object4 NodeRecord
+        ("tag" := Json.Decode.string)
+        ("children" := Json.Decode.list (Json.Decode.at [ "_1" ] decodeNodeType))
```

Comment:
Please pull `Json.Decode.at [ "_1" ] decodeNodeType` out into it's own decoder with a name that explains what you're pulling out and wherefrom (e.g, why don't we care about the other fields in this object?)